docs/decorator-reference.md

```diff
@@ -374,6 +375,36 @@ export class User {
 
 Value will be generated only once, before inserting the entity into the database.
 
+#### `@VirtualColumn`
+
+Special column that is never saved to the database and thus acts as a readonly property.
+Each time you call `find` or `findOne` from the entity manager, the value is recalculated based on the query function that was provided in the VirtualColumn Decorator. The alias argument passed to the query references the exact entity alias of the generated query behind the scenes.
+
+```typescript
+@Entity({ name: "companies", alias: "COMP" })
+export class Company extends BaseEntity {
+  @PrimaryColumn("varchar", { length: 50 })
+  name: string;
+
+  @VirtualColumn({ query: (alias) => `SELECT COUNT("name") FROM "employees" WHERE "companyName" = ${alias}.name` })
+  totalEmployeesCount: number;
```

Comment:
BTW, are users allowed to:

`.getRepository(Company).find({where: {totalEmployeesCount: Equal(5)})`?

Reply:
Yes, users are allowed to do this. I made sure that you could perform where & order by's with the entity manager. You can view my test case here: 

I did however run into a "limitation" (limitation but there is a workaround) with "Wheres" and "Order Bys" with the query builder. They would have to make a generic function that returns the virtual column as a query string and export that function so they could generate the query within the query builder.

Reply:
Specifically this:

```js
                 where: {
                        name: companyName,
                        totalEmployeesCount: MoreThan(2),
                        employees: {
                            timesheets: {
                                totalActvityHours: MoreThan(0),
                            },
                        },
                    },
                    order: {
                        employees: {
                            timesheets: {
                                id: "DESC",
                                totalActvityHours: "ASC",
                            },
                        },
                    },
```